commands-yml/commands/device/network/network-speed.yml

```diff
@@ -0,0 +1,68 @@
+---
+name: Network speed
+short_description: Set network speed (Emulator only)
+
+example_usage:
+  java:
+    |
+      driver.setNetworkSpeed(NetworkSpeed.LTE);
+  python:
+    |
+      self.driver.set_network_speed(NetSpeed.LTE)
+  javascript_wd:
+    |
+      await driver.networkSpeed('Network type');
+  javascript_wdio:
+    |
+      driver.networkSpeed('lte');
+  ruby:
+    |
+      set_network_speed :lte
+  ruby_core:
+    |
+      @driver.set_network_speed :lte
+
+client_docs:
+  java: "https://appium.github.io/java-client/io/appium/java_client/android/SupportsSpecialEmulatorCommands.html#setNetworkSpeed-io.appium.java_client.android.NetworkSpeed-"
+  python: "https://github.com/appium/python-client"
+  javascript_wd: "https://github.com/admc/wd/blob/master/lib/commands.js#L3411"
+  ruby: "https://www.rubydoc.info/github/appium/ruby_lib_core/master/Appium/Core/Android/Device/Emulator#set_network_speed-instance_method"
+  ruby_core: "https://www.rubydoc.info/github/appium/ruby_lib_core/master/Appium/Core/Android/Device/Emulator#set_network_speed-instance_method"
+
+# Driver support by platform
+driver_support:
+  ios:
+    xcuitest: false
+    uiautomation: false
+  android:
+    uiautomator2: false
```

Comment:
Both UIA2 and Espresso can work if the test device is emulator since the change is via adb command in preparing emulator function
https://github.com/appium/appium-android-driver/blob/14b8684a3bf145ee390a2155104af5661a2ec36d/lib/android-helpers.js#L130

Reply:
Yeah, that's true. I've just copied here from gsm-voice.
I'll update not only here but also others related to emulator function.